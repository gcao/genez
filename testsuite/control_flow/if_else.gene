#!/usr/bin/env gene run

# Test: If-else conditionals
(fn test_if []
  (if true
    (print "True branch taken")
  else
    (print "This should not print"))
  
  (if false
    (print "This should not print")
  else
    (print "False branch taken"))
  
  # Nested conditionals
  (var x 10)
  (if (< 5 x)
    (if (< x 15)
      (print "x is between 5 and 15")
    else
      (print "x is not less than 15"))
  else
    (print "x is not greater than 5")))

(test_if)