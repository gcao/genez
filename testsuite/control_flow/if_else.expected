True branch taken
False branch taken
x is between 5 and 15